### PR TITLE
added asg reference.

### PR DESCRIPTION
**Pull request by siennathesane**
removed the custom route service documentation in favour of asgs.

Signed-off-by: Mike Lloyd <mike.lloyd@gsa.gov>

### CODE REVIEW

File: content/docs/apps/experimental/restricting-users-to-trusted-ip-ranges.md [hunk 1]
```diff
@@ -20,5 +20,60 @@ To fully limit access to your apps and services, you must ensure that all users
 
 ## Restricting access to your own applications
 
-To limit requests to your own applications, you can modify your application logic directly. Alternatively, you can [can follow this example](https://github.com/alphagov/re-paas-ip-safelist-service#quick-demo) within cloud.gov to [create a route service](https://docs.cloudfoundry.org/services/route-services.html#introduction) 
-to block all traffic that does not originate from a trusted IP address range.
+Limiting or restricting access for both ingress and egress traffic from your applications is fairly straightforward through a feature called [Application Security Groups](https://docs.cloudfoundry.org/concepts/asg.html), or ASGs. ASGs support white- and black-listing network traffic from given IPv4 ranges.
+
+### Scoping
```
Comment (mogul) [inline content/docs/apps/experimental/restricting-users-to-trusted-ip-ranges.md:25]: I'd cut this section on scoping. It's not very relevant to customers, and they can read about it in the CF docs if they need to.

File: content/docs/apps/experimental/restricting-users-to-trusted-ip-ranges.md [hunk 2]
```diff
@@ -20,5 +20,60 @@ To fully limit access to your apps and services, you must ensure that all users
...
+
+Application Security Groups (ASGs) support two types of scoping:
+
+* Platform
+* Space
+
+These scopes allow both operators and developers to have more fine grained access control. Platform-scoped ASGs are applied at the platform level, and would apply to all tenants of cloud.gov. The cloud.gov team does not apply platform-scoped ASGs as they can interrupt tenant operations unless they are needed to meet platform compliance requirements.
+
+Space-scoped ASGs are specific to a given tenant's Space within their allotted organization. This allows for tenants to create Space-specific security groups that apply to all applications within a given space. To apply a Space-scoped ASG to a single application, create a new Space, assign the ASG, and then deploy the application.
```
Comment (mogul) [inline content/docs/apps/experimental/restricting-users-to-trusted-ip-ranges.md]: ...except this line about space-scoped ASGs, keep this one! Also s/Space/space/.

File: content/docs/apps/experimental/restricting-users-to-trusted-ip-ranges.md [hunk 3]
```diff
@@ -20,5 +20,60 @@ To fully limit access to your apps and services, you must ensure that all users
 
 ## Restricting access to your own applications
 
-To limit requests to your own applications, you can modify your application logic directly. Alternatively, you can [can follow this example](https://github.com/alphagov/re-paas-ip-safelist-service#quick-demo) within cloud.gov to [create a route service](https://docs.cloudfoundry.org/services/route-services.html#introduction) 
-to block all traffic that does not originate from a trusted IP address range.
+Limiting or restricting access for both ingress and egress traffic from your applications is fairly straightforward through a feature called [Application Security Groups](https://docs.cloudfoundry.org/concepts/asg.html), or ASGs. ASGs support white- and black-listing network traffic from given IPv4 ranges.
```
Comment (mogul) [inline content/docs/apps/experimental/restricting-users-to-trusted-ip-ranges.md]: I'd rework this paragraph to talk about the two cases... "If you only need to restrict to IPv4 ranges, use ASGs. For IPv6 and other more complex restriction cases (eg limiting particular kinds of requests), you can use a proxy app". Then you can add an anchored section heading at the bottom which gives them a pointer to making a front-end proxy app with the nginx-buildpack.

I believe Peter wrote an example of doing this, btw, so we might even have an something to point them to! See [Slack discussion](https://gsa-tts.slack.com/archives/C0DF6LAMQ/p1553803321013900).

File: content/docs/apps/experimental/restricting-users-to-trusted-ip-ranges.md [hunk 4]
```diff
@@ -20,5 +20,60 @@ To fully limit access to your apps and services, you must ensure that all users
...
+These scopes allow both operators and developers to have more fine grained access control. Platform-scoped ASGs are applied at the platform level, and would apply to all tenants of cloud.gov. The cloud.gov team does not apply platform-scoped ASGs as they can interrupt tenant operations unless they are needed to meet platform compliance requirements.
+
+Space-scoped ASGs are specific to a given tenant's Space within their allotted organization. This allows for tenants to create Space-specific security groups that apply to all applications within a given space. To apply a Space-scoped ASG to a single application, create a new Space, assign the ASG, and then deploy the application.
+
+### Defaults
+
+cloud.gov is preconfigured with two ASGs: `public_networks` and `dns`. These ASGs are applied by default to all containers in your deployment.
+
+* `public_networks`: This group allows access to public networks, and blocks access to private networks and link-local addresses. Cloud Foundry blocks outgoing traffic to the following IP address ranges by specifically allowing traffic to all other addresses.
+
```
Comment (mogul) [inline content/docs/apps/experimental/restricting-users-to-trusted-ip-ranges.md]: Believe it or not, compliance rules forbid us to publish internal network CIDRs in docs even when they're private ranges. 😞 So better to just say that private IP ranges are blocked and let our readers look up what private ranges are, removing the following lines.

**Review by mogul — CHANGES_REQUESTED** [review]
Sorry for the hassle, want to be thorough and unfortunately less-thorough due to compliance here. @pburkholder is your Trello proxy published anywhere that we can link to?

File: content/docs/apps/experimental/restricting-users-to-trusted-ip-ranges.md (hunk 1 above)
Comment (siennathesane) [reply content/docs/apps/experimental/restricting-users-to-trusted-ip-ranges.md:25]: I only cover scoping so I can specify that we do not deploy platform-scoped ASGs - I think it's worth leaving in there. I did want to specify that we will not accept requests for platform-scoped policies but that seems a bit aggressive for the docs.

**Review by mogul — DISMISSED** [review]

**Review by mogul — APPROVED** [review]

**Pull request merged**